pyaedt/edb_core/EDB_Data.py

```diff
@@ -2663,9 +2663,32 @@ def refdes(self):
         """
         return self.edbcomponent.GetName()
 
+    @property
+    def is_enabled(self):
+        """Check if the current object is enabled.
+
+        Returns
+        -------
+        bool
+            ``True`` if current object is enabled, ``False`` otherwise.
+        """
+        return self.edbcomponent.GetComponentProperty().IsEnabled()
+
+    @is_enabled.setter
+    def is_enabled(self, enabled):
+        """Enables the current object."""
+        if self.edbcomponent.GetComponentType() in [
+            self._edb.Definition.ComponentType.Resistor,
+            self._edb.Definition.ComponentType.Capacitor,
+            self._edb.Definition.ComponentType.Inductor,
+        ]:
+            return self.edbcomponent.GetComponentProperty().SetEnabled(enabled)
```

Comment:
@ring630 .
This is causing the IronPython tests failure you were asking about.
IronPython is less permissive that Python with some incorrect types provided.
![image](https://user-images.githubusercontent.com/87315832/178173735-74c4f57d-af58-41cd-9e35-91ede3b5a39f.png)

You can debug in Python on your local machine if it is more convenient for you and check the type returned by `self.edbcomponent.GetComponentProperty()`.

Reply:
@MaxJPRey 
GetComponentProperty() returns RLCComponentProperty on my machine.

Reply:
@ring630 I fixed the issue. Actually with the new version Python behaves exactly the same as the EDB API with C#. I don't understand quite well how python was working before. If you want to access some properties of object (e.g ComponentProperty for instance) you need to clone the object to make it not read only. Then you will have access to all methods. I fixed all getter and setters in the edb components. Now the test is passing